src/fstab-generator/fstab-generator.c

```diff
@@ -763,21 +763,52 @@ static int parse_fstab(bool initrd) {
                         flags |= rw_only * MOUNT_RW_ONLY |
                                  automount * MOUNT_AUTOMOUNT;
 
+                        is_sysroot = path_equal(where, "/sysroot");
+                        /* See comment from add_sysroot_usr_mount about the need for extra indirection
+                         * in case /usr needs to be mounted in order for the root fs to be synthesized
+                         * based on configuration included in /usr/, e.g. systemd-repart. */
+                        is_sysroot_usr = path_equal(where, "/sysroot/usr");
```

Comment:
I think this should be conditionalized on `in_initrd()` (i.e. the function by that name, not the function parameter `initrd` we receive here). i.e. on the host /sysroot/ should really not be a thing